src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        result += this[i];
+      }
+
+      if (i === this.length - 1) {
+        break;
+      }
+      result += separator;
```

Comment:
```suggestion
      if (i !== this.length - 1) {
        result += separator;
      }
```
Just add separator only if it's not the last iteration.
It will make your code a bit cleaner